sns_driver_test.go

```diff
@@ -0,0 +1,129 @@
+package es_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/indebted-modules/es"
+	"github.com/stretchr/testify/suite"
+)
+
+type SNSNotifierSuite struct {
+	suite.Suite
+	snsSvc   *sns.SNS
+	sqsSvc   *sqs.SQS
+	topicArn *string
+	queueURL *string
+}
+
+func TestSNSNotifierSuite(t *testing.T) {
+	suite.Run(t, new(SNSNotifierSuite))
+}
+
+func (s *SNSNotifierSuite) SetupSuite() {
+	snsEndpoint := "http://localstack:4575"
+	sqsEndpoint := "http://localstack:4576"
+	s.Eventually(func() bool {
+		_, err1 := http.Get(snsEndpoint)
+		_, err2 := http.Get(sqsEndpoint)
+		return err1 == nil && err2 == nil
+	}, 10*time.Second, time.Second, "Localstack services are not ready or running")
```

Comment:
Maybe we could leverage this approach for other tests that have an external dependency? Although it adds a bit more to the test setup, it removes the need for something like our `make db.is-up` kind of check.

Reply:
To be honest I'm not a huge fan of either option we have so far, but I also lack the context on why we did it this way. Keen for a chat this week to explore potential alternatives.